pkgs/native_assets_cli/test/locking/locking_test.dart

```diff
@@ -131,13 +131,27 @@ void main() async {
             if (timeout != null) timeout.inMilliseconds.toString(),
           ],
         );
+
+        final stdoutSub = process.stdout
+            .transform(utf8.decoder)
+            .transform(const LineSplitter())
+            .listen(logger.fine);
+        final stderrSub = process.stderr
+            .transform(utf8.decoder)
+            .transform(const LineSplitter())
+            .listen(logger.severe);
+
+        final (exitCode, _, _) = await (
+          process.exitCode,
+          stdoutSub.asFuture<void>(),
```

Comment:
Why not just wait for the exit code?

Reply:
It can complete before the output stream is finished, which leads to dropped logs.